src/HealthChecks.AzureStorage/AzureBlobStorageHealthCheck.cs

```diff
@@ -7,79 +6,38 @@ namespace HealthChecks.AzureStorage
+                  ClientCache<BlobServiceClient>.GetOrAdd(connectionString, k => new BlobServiceClient(k, clientOptions)),
+                  new AzureBlobStorageHealthCheckOptions { ContainerName = containerName })
+        { }
+
+        public AzureBlobStorageHealthCheck(Uri blobServiceUri, TokenCredential credential, string? containerName = default, BlobClientOptions? clientOptions = default)
+            : this(
+                  ClientCache<BlobServiceClient>.GetOrAdd(blobServiceUri?.ToString()!, _ => new BlobServiceClient(blobServiceUri, credential, clientOptions)),
+                  new AzureBlobStorageHealthCheckOptions { ContainerName = containerName })
+        { }
+
+        public AzureBlobStorageHealthCheck(BlobServiceClient blobServiceClient, AzureBlobStorageHealthCheckOptions options)
         {
-            _connectionString = connectionString ?? throw new ArgumentNullException(nameof(connectionString));
-            _containerName = containerName;
-            _clientOptions = clientOptions;
-        }
-
-        public AzureBlobStorageHealthCheck(Uri blobServiceUri, TokenCredential credential, string? containerName = default, BlobClientOptions? clientOptions = null)
-        {
-            _blobServiceUri = blobServiceUri ?? throw new ArgumentNullException(nameof(blobServiceUri));
-            _azureCredential = credential ?? throw new ArgumentNullException(nameof(credential));
-            _containerName = containerName;
-            _clientOptions = clientOptions;
+            _blobServiceClient = blobServiceClient ?? throw new ArgumentNullException(nameof(blobServiceClient));
+            _options = options ?? throw new ArgumentNullException(nameof(options));
         }
 
         public async Task<HealthCheckResult> CheckHealthAsync(HealthCheckContext context, CancellationToken cancellationToken = default)
         {
-            try
-            {
-                var blobServiceClient = GetBlobServiceClient();
-                await foreach (var page in blobServiceClient.GetBlobContainersAsync(cancellationToken: cancellationToken).AsPages(pageSizeHint: 1))
-                {
-                    break;
-                }
-
-                if (!string.IsNullOrEmpty(_containerName))
-                {
-                    var containerClient = blobServiceClient.GetBlobContainerClient(_containerName);
-
-                    if (!await containerClient.ExistsAsync(cancellationToken))
-                    {
-                        return new HealthCheckResult(context.Registration.FailureStatus, description: $"Container '{_containerName}' not exists");
-                    }
-
-                    await containerClient.GetPropertiesAsync(cancellationToken: cancellationToken);
-                }
-
-                return HealthCheckResult.Healthy();
-            }
-            catch (Exception ex)
-            {
-                return new HealthCheckResult(context.Registration.FailureStatus, exception: ex);
-            }

```

Comment:
Why did you remove this? This is a "standard" pattern for all HCs across repo.

Reply:
The same note for other HCs.

Reply:
I mentioned it briefly in the PR description, but this try/catch block is superfluous. The health check is invoked from the underlying `DefaultHealthCheckService` and that class will catch any unhandled exceptions and return unhealthy. The only difference is that it will use the exception's message as the message for the result. If we don't want that behavior, I can change it back. I just thought for simplicity's sake we could remove it

Reply:
Indeed. It **can** be superfluous - for **particular** implementation (`DefaultHealthCheckService`). Let's keep try/catch blocks as before. You can post a new issue about removing all try/catch blocks inside HCs across this repo in separate PRif you would like.

Reply:
Restored the outer-try/catch